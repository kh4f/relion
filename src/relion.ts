import { bump, commit, tag, changelog } from '@/lifecycles'
import { resolveConfig, setSilent } from '@/utils'
import type { UserConfig } from '@/types'

export default async function relion(userConfig: UserConfig) {
	setSilent(!!userConfig.silent || (!!userConfig.profile && !!userConfig[`_${userConfig.profile}`]?.silent))

	const resolvedConfig = await resolveConfig(userConfig)

	const bumpResults = bump(resolvedConfig)
	const generatedChangelog = changelog(resolvedConfig)
	const commitCommand = commit(resolvedConfig)
	const tagCommand = tag(resolvedConfig)

	return { resolvedConfig, generatedChangelog, commitCommand, tagCommand, bumpResults }
}

export const defineConfig = (config: UserConfig): UserConfig => config