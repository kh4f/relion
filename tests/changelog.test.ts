import { describe, expect, it } from 'vitest'
import relion from '@/.'
import { testConfig } from './fixtures/relion.test-config'

describe('changelog generation', () => {
	it('should print upcoming release changelog', () => {
		relion({ ...testConfig })
	})

	it('should print latest release changelog', () => {
		relion({ ...testConfig,
			changelog: {
				commitRange: 'latest-release',
			},
		})
	})

	it('should print changelog for last 5 commits', () => {
		relion({ ...testConfig,
			changelog: {
				commitRange: 'HEAD~5..',
			},
		})
	})

	it.for(['v0.7.0', 'v0.8.0'])('should print changelog for release $0', (releaseTag) => {
		const result = relion({ ...testConfig,
			changelog: {
				commitRange: { releaseTag },
			},
		})
		expect(result.generatedChangelog).toMatchSnapshot()
	})
})