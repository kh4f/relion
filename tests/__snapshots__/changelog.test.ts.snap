// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`changelog generation > should generate changelog for release 'v0.7.0' 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.7.0  \`](https://github.com/kh4f/relion/compare/v0.6.0...v0.7.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **core**: add \`defaultChangelogSections\` export [\`86da78a\`](https://github.com/kh4f/relion/commit/86da78a)
- **changelog**: extend section definitions with optional \`id\` property [\`41c838a\`](https://github.com/kh4f/relion/commit/41c838a)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **templates**: use \`id\` property instead of \`commitType\` for breaking changes condition [\`205b9f8\`](https://github.com/kh4f/relion/commit/205b9f8)
- **defaults**: update \`prevReleaseTagPattern\` regex to allow optional leading \`v\` [\`1c631eb\`](https://github.com/kh4f/relion/commit/1c631eb)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üìö Documentation
- **commit-naming**: update commit message guidelines with clarified rules and examples [\`94ca4fc\`](https://github.com/kh4f/relion/commit/94ca4fc)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üé® Style
- **changelog**: update title for default \`style\` section from 'Formatting' to 'Style' [\`d47d472\`](https://github.com/kh4f/relion/commit/d47d472)
- **templates**: 
  - remove emoji from footer commit link [\`7ea3c1d\`](https://github.com/kh4f/relion/commit/7ea3c1d)
  - update release footer with italic link and \`‚Ä¢\` separator [\`7e56b38\`](https://github.com/kh4f/relion/commit/7e56b38)
  - simplify release footer by omitting \`prevTag ‚Üí tag\` [\`df84af2\`](https://github.com/kh4f/relion/commit/df84af2)
  - update footer link label to 'All Release Commits' [\`2ca80a6\`](https://github.com/kh4f/relion/commit/2ca80a6)
  - update commit scope formatting to bold with colon [\`37dcb53\`](https://github.com/kh4f/relion/commit/37dcb53)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üõ†Ô∏è Chores
- **renovate**: add initial configuration for Renovate [\`bc296e9\`](https://github.com/kh4f/relion/commit/bc296e9)

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.6.0...v0.7.0) &ensp;‚Ä¢&ensp; _Sep 11, 2025_


"
`;

exports[`changelog generation > should generate changelog for release 'v0.7.0' without scope groups 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.7.0  \`](https://github.com/kh4f/relion/compare/v0.6.0...v0.7.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **core**: add \`defaultChangelogSections\` export [\`86da78a\`](https://github.com/kh4f/relion/commit/86da78a)
- **changelog**: extend section definitions with optional \`id\` property [\`41c838a\`](https://github.com/kh4f/relion/commit/41c838a)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **templates**: use \`id\` property instead of \`commitType\` for breaking changes condition [\`205b9f8\`](https://github.com/kh4f/relion/commit/205b9f8)
- **defaults**: update \`prevReleaseTagPattern\` regex to allow optional leading \`v\` [\`1c631eb\`](https://github.com/kh4f/relion/commit/1c631eb)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üìö Documentation
- **commit-naming**: update commit message guidelines with clarified rules and examples [\`94ca4fc\`](https://github.com/kh4f/relion/commit/94ca4fc)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üé® Style
- **changelog**: update title for default \`style\` section from 'Formatting' to 'Style' [\`d47d472\`](https://github.com/kh4f/relion/commit/d47d472)
- **templates**: remove emoji from footer commit link [\`7ea3c1d\`](https://github.com/kh4f/relion/commit/7ea3c1d)
- **templates**: update release footer with italic link and \`‚Ä¢\` separator [\`7e56b38\`](https://github.com/kh4f/relion/commit/7e56b38)
- **templates**: simplify release footer by omitting \`prevTag ‚Üí tag\` [\`df84af2\`](https://github.com/kh4f/relion/commit/df84af2)
- **templates**: update footer link label to 'All Release Commits' [\`2ca80a6\`](https://github.com/kh4f/relion/commit/2ca80a6)
- **templates**: update commit scope formatting to bold with colon [\`37dcb53\`](https://github.com/kh4f/relion/commit/37dcb53)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üõ†Ô∏è Chores
- **renovate**: add initial configuration for Renovate [\`bc296e9\`](https://github.com/kh4f/relion/commit/bc296e9)

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.6.0...v0.7.0) &ensp;‚Ä¢&ensp; _Sep 11, 2025_


"
`;

exports[`changelog generation > should generate changelog for release 'v0.8.0' 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.8.0  \`](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- The \`defaultChangelogSections\` export has been removed from the public API. Use \`changelogSectionsSelector\` instead, which provides the same sections plus \`pick\` and \`omit\` methods. <sup>[1]</sup>
- The \`sections\` option in changelog configuration is now an object with section IDs as keys instead of an array. <sup>[2]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **utils**: add \`changelogSectionsSelector\` with \`pick\` and \`omit\` methods [\`3febcbc\`](https://github.com/kh4f/relion/commit/3febcbc) ‚ö†Ô∏è<sup>[1]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **changelog**: 
  - change \`sections\` from array to object format for better usability [\`349fed9\`](https://github.com/kh4f/relion/commit/349fed9) ‚ö†Ô∏è<sup>[2]</sup>
  - rename \`commitGroups\` property to \`commitTypeGroups\` [\`6ec89a8\`](https://github.com/kh4f/relion/commit/6ec89a8)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üè∑Ô∏è Types
- **defaults**: use \`satisfies ChangelogSectionsMap\` for \`defaultChangelogSections\` [\`4343e6f\`](https://github.com/kh4f/relion/commit/4343e6f)

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0) &ensp;‚Ä¢&ensp; _Sep 12, 2025_


"
`;

exports[`changelog generation > should generate changelog for release 'v0.8.0' without scope groups 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.8.0  \`](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- The \`defaultChangelogSections\` export has been removed from the public API. Use \`changelogSectionsSelector\` instead, which provides the same sections plus \`pick\` and \`omit\` methods. <sup>[1]</sup>
- The \`sections\` option in changelog configuration is now an object with section IDs as keys instead of an array. <sup>[2]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **utils**: add \`changelogSectionsSelector\` with \`pick\` and \`omit\` methods [\`3febcbc\`](https://github.com/kh4f/relion/commit/3febcbc) ‚ö†Ô∏è<sup>[1]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **changelog**: change \`sections\` from array to object format for better usability [\`349fed9\`](https://github.com/kh4f/relion/commit/349fed9) ‚ö†Ô∏è<sup>[2]</sup>
- **changelog**: rename \`commitGroups\` property to \`commitTypeGroups\` [\`6ec89a8\`](https://github.com/kh4f/relion/commit/6ec89a8)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üè∑Ô∏è Types
- **defaults**: use \`satisfies ChangelogSectionsMap\` for \`defaultChangelogSections\` [\`4343e6f\`](https://github.com/kh4f/relion/commit/4343e6f)

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0) &ensp;‚Ä¢&ensp; _Sep 12, 2025_


"
`;

exports[`changelog generation > should generate changelog with release changelog URL in footer 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.18.0  \`](https://github.com/kh4f/relion/compare/v0.17.1...v0.18.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- The \`versionSourceFile\` is no longer automatically included in the \`bump\` array (but still is if \`bump\` is \`true\`). <sup>[1]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ôªÔ∏è Reverts
- **config**: allow skipping bump for \`versionSourceFile\` via \`!versionSourceFile\` in \`bump\` array [\`0ac5ff0\`](https://github.com/kh4f/relion/commit/0ac5ff0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **config**: require explicit \`versionSourceFile\` inclusion in \`bump\` array [\`fd5ad3c\`](https://github.com/kh4f/relion/commit/fd5ad3c) ‚ö†Ô∏è<sup>[1]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üé® Style
- **ci/release**: inline \`with\` block for Node.js setup step [\`a0d7905\`](https://github.com/kh4f/relion/commit/a0d7905)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üß™ Tests
- **changelog**: 
  - skip test for generating changelog with transformed partials [\`9c064b7\`](https://github.com/kh4f/relion/commit/9c064b7)
  - update snapshot for transformed partials test [\`67ed389\`](https://github.com/kh4f/relion/commit/67ed389)
  - update snapshot for changelog generation tests [\`a9dc9a2\`](https://github.com/kh4f/relion/commit/a9dc9a2)
- **bump**: add test for bumping version in \`manifest.json\` [\`f2646f3\`](https://github.com/kh4f/relion/commit/f2646f3)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöÄ CI
- **checks**: improve comment clarity for commit history fetching [\`d787525\`](https://github.com/kh4f/relion/commit/d787525)
- **workflows**: update lint step to use \`lint:fix\` script [\`1aff340\`](https://github.com/kh4f/relion/commit/1aff340)
- **release**: add linting and testing steps to release workflow [\`b61b08d\`](https://github.com/kh4f/relion/commit/b61b08d)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üß© Dependencies
- **dev-deps**: bump \`lint-staged\` from 16.2.1 to 16.2.3 (#9) [\`0ae604d\`](https://github.com/kh4f/relion/commit/0ae604d)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üõ†Ô∏è Chores
- **renovate**: replace \`matchCurrentAge\` with \`minimumReleaseAge\` for patch and minor updates [\`9471d0d\`](https://github.com/kh4f/relion/commit/9471d0d)

##### &emsp;&ensp;&nbsp;&nbsp; [_Release Changelog_](https://github.com/kh4f/relion/blob/main/CHANGELOG.md#--v0180-) &ensp;‚Ä¢&ensp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.17.1...v0.18.0) &ensp;‚Ä¢&ensp; _Oct 2, 2025_


"
`;

exports[`changelog generation > should generate changelog with scope groups for custom commits 1`] = `
"### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- unscoped commit 2 fake_269b30b
- unscoped commit 1 fake_b86c57f
- **scope 2**: 
  - scoped commit 3 fake_728ea8b
  - scoped commit 2 fake_fdf4bef
- **scope 1**: scoped commit 1 fake_1bf47c1

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ü©π Fixes
- unscoped commit 3 fake_923cc1a

"
`;

exports[`changelog generation > should generate changelog with selected and modified sections 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.7.0  \`](https://github.com/kh4f/relion/compare/v0.6.0...v0.7.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üéÅ New Features
- **core**: add \`defaultChangelogSections\` export [\`86da78a\`](https://github.com/kh4f/relion/commit/86da78a)
- **changelog**: extend section definitions with optional \`id\` property [\`41c838a\`](https://github.com/kh4f/relion/commit/41c838a)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **templates**: use \`id\` property instead of \`commitType\` for breaking changes condition [\`205b9f8\`](https://github.com/kh4f/relion/commit/205b9f8)
- **defaults**: update \`prevReleaseTagPattern\` regex to allow optional leading \`v\` [\`1c631eb\`](https://github.com/kh4f/relion/commit/1c631eb)

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.6.0...v0.7.0) &ensp;‚Ä¢&ensp; _Sep 11, 2025_


"
`;

exports[`changelog line limit > should keep commits with breaking changes 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.9.0  \`](https://github.com/kh4f/relion/compare/v0.8.0...v0.9.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- some breaking change <sup>[1]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **core**: some feature fake_4271cd6

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ü©π Fixes
- **core**: some bugfix fake_5087d7d

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö° Performance
- **core**: some performance improvement fake_27db403

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ôªÔ∏è Reverts
- **core**: some revert fake_8bf8ff8

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üõ†Ô∏è Chores
- **core**: some chore fake_678a37e ‚ö†Ô∏è<sup>[1]</sup>

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.8.0...v0.9.0) &ensp;‚Ä¢&ensp; __


"
`;

exports[`changelog line limit > should keep sections with ignoreLimit=true despite changelog line limit 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.9.0  \`](https://github.com/kh4f/relion/compare/v0.8.0...v0.9.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **core**: 
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6
  - some feature fake_4271cd6

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ü©π Fixes
- **core**: 
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d
  - some bugfix fake_5087d7d

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö° Performance
- **core**: some performance improvement fake_27db403

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ôªÔ∏è Reverts
- **core**: some revert fake_8bf8ff8

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.8.0...v0.9.0) &ensp;‚Ä¢&ensp; __


"
`;

exports[`changelog line limit > should limit changelog to 10 lines for release v0.17.0 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.17.0  \`](https://github.com/kh4f/relion/compare/v0.16.0...v0.17.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- The \`VersionedFile\` type now uses \`file\` and \`pattern\` properties instead of \`filePath\` and \`versionPattern\`. <sup>[1]</sup>
- The \`silent\` option has been removed. Use \`logLevel\` instead to control logging verbosity. <sup>[2]</sup>
- The \`bumpResults\` property is no longer returned by the main \`relion()\` function. <sup>[3]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **changelog**: add function support for \`partials\` to transform fallback content [\`7f070fd\`](https://github.com/kh4f/relion/commit/7f070fd)
- **bump**: add support for custom replacement for versioned files [\`77aa591\`](https://github.com/kh4f/relion/commit/77aa591)
- **config**: 
  - allow skipping bump for \`versionSourceFile\` via \`!versionSourceFile\` in \`bump\` array [\`ba77ad8\`](https://github.com/kh4f/relion/commit/ba77ad8)
  - support extracting current version from latest release tag instead of file [\`9d96a36\`](https://github.com/kh4f/relion/commit/9d96a36)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ü©π Fixes
- **config-resolver**: preserve all default versioned file attributes when matching custom file [\`33aa299\`](https://github.com/kh4f/relion/commit/33aa299)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **config**: 
  - rename \`filePath\` and \`versionPattern\` to \`file\` and \`pattern\` [\`88f1446\`](https://github.com/kh4f/relion/commit/88f1446) ‚ö†Ô∏è<sup>[1]</sup>
  - replace \`silent\` option with \`logLevel\` for improved logging control [\`9cfb8af\`](https://github.com/kh4f/relion/commit/9cfb8af) ‚ö†Ô∏è<sup>[2]</sup>
- **core**: remove \`bumpResults\` from \`relion()\` return values [\`0626642\`](https://github.com/kh4f/relion/commit/0626642) ‚ö†Ô∏è<sup>[3]</sup>

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.16.0...v0.17.0) &ensp;‚Ä¢&ensp; _Sep 28, 2025_


"
`;

exports[`changelog line limit > should limit changelog to 20 lines for release v0.17.0 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.17.0  \`](https://github.com/kh4f/relion/compare/v0.16.0...v0.17.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- The \`VersionedFile\` type now uses \`file\` and \`pattern\` properties instead of \`filePath\` and \`versionPattern\`. <sup>[1]</sup>
- The \`silent\` option has been removed. Use \`logLevel\` instead to control logging verbosity. <sup>[2]</sup>
- The \`bumpResults\` property is no longer returned by the main \`relion()\` function. <sup>[3]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **changelog**: add function support for \`partials\` to transform fallback content [\`7f070fd\`](https://github.com/kh4f/relion/commit/7f070fd)
- **bump**: add support for custom replacement for versioned files [\`77aa591\`](https://github.com/kh4f/relion/commit/77aa591)
- **config**: 
  - allow skipping bump for \`versionSourceFile\` via \`!versionSourceFile\` in \`bump\` array [\`ba77ad8\`](https://github.com/kh4f/relion/commit/ba77ad8)
  - support extracting current version from latest release tag instead of file [\`9d96a36\`](https://github.com/kh4f/relion/commit/9d96a36)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ü©π Fixes
- **config-resolver**: preserve all default versioned file attributes when matching custom file [\`33aa299\`](https://github.com/kh4f/relion/commit/33aa299)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **utils**: extract \`hbs-load-hook\` to separate module to avoid registration conflict [\`88c6fea\`](https://github.com/kh4f/relion/commit/88c6fea)
- **release**: update relion config to use 'logLevel' [\`591e2c8\`](https://github.com/kh4f/relion/commit/591e2c8)
- **config**: 
  - rename \`filePath\` and \`versionPattern\` to \`file\` and \`pattern\` [\`88f1446\`](https://github.com/kh4f/relion/commit/88f1446) ‚ö†Ô∏è<sup>[1]</sup>
  - replace \`silent\` option with \`logLevel\` for improved logging control [\`9cfb8af\`](https://github.com/kh4f/relion/commit/9cfb8af) ‚ö†Ô∏è<sup>[2]</sup>
- **logger**: remove unused \`deepLog\` function [\`9c0b9a2\`](https://github.com/kh4f/relion/commit/9c0b9a2)
- **commits-parser**: replace \`console.warn\` with \`warn\` utility function [\`9982505\`](https://github.com/kh4f/relion/commit/9982505)
- **core**: remove \`bumpResults\` from \`relion()\` return values [\`0626642\`](https://github.com/kh4f/relion/commit/0626642) ‚ö†Ô∏è<sup>[3]</sup>
- **vitest**: use \`silent: true\` in config instead of mocking \`console.log\` [\`9cb40a7\`](https://github.com/kh4f/relion/commit/9cb40a7)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üé® Style
- **config-resolver**: add quotes around \`versionSourceFile.file\` in log output [\`7f08be2\`](https://github.com/kh4f/relion/commit/7f08be2)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üß™ Tests
- **changelog**: wrap tests for partials customization in dedicated describe block [\`7272cf7\`](https://github.com/kh4f/relion/commit/7272cf7)
- **fixtures**: add fixture files for \`manifest.json\`, \`package.json\`, and \`versions.json\` [\`dc46f7c\`](https://github.com/kh4f/relion/commit/dc46f7c)
- **setup**: mock \`defaults\` module for configurable \`dryRun\` and \`changelog.output\` in tests [\`c7ba9dc\`](https://github.com/kh4f/relion/commit/c7ba9dc)

##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.16.0...v0.17.0) &ensp;‚Ä¢&ensp; _Sep 28, 2025_


"
`;

exports[`commit references rendering > should render commit references as hyperlinks 1`] = `
"### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **core**: add new feature [\`123456b\`](https://github.com/kh4f/relion/commit/123456b) ([#20](https://github.com/kh4f/relion/issues/20), [kh4f/relion#30](https://github.com/kh4f/relion/issues/30))

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ü©π Fixes
- **core**: fix bug [\`123456a\`](https://github.com/kh4f/relion/commit/123456a) ([#10](https://github.com/kh4f/relion/issues/10))

"
`;

exports[`commit references rendering > should render commit references as plain text 1`] = `
"### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **core**: add new feature 123456b (#20, kh4f/relion#30)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ü©π Fixes
- **core**: fix bug 123456a (#10)

"
`;

exports[`partials customization > should generate changelog with customized partials 1`] = `"CUSTOM FOOTER --- **This is a custom partial** ---"`;

exports[`partials customization > should generate changelog with transformed footer and custom changelog URL partial 1`] = `
"# Changelog


## &ensp; [\` üì¶ v0.8.0  \`](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- The \`defaultChangelogSections\` export has been removed from the public API. Use \`changelogSectionsSelector\` instead, which provides the same sections plus \`pick\` and \`omit\` methods. <sup>[1]</sup>
- The \`sections\` option in changelog configuration is now an object with section IDs as keys instead of an array. <sup>[2]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **utils**: add \`changelogSectionsSelector\` with \`pick\` and \`omit\` methods [\`3febcbc\`](https://github.com/kh4f/relion/commit/3febcbc) ‚ö†Ô∏è<sup>[1]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **changelog**: 
  - change \`sections\` from array to object format for better usability [\`349fed9\`](https://github.com/kh4f/relion/commit/349fed9) ‚ö†Ô∏è<sup>[2]</sup>
  - rename \`commitGroups\` property to \`commitTypeGroups\` [\`6ec89a8\`](https://github.com/kh4f/relion/commit/6ec89a8)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üè∑Ô∏è Types
- **defaults**: use \`satisfies ChangelogSectionsMap\` for \`defaultChangelogSections\` [\`4343e6f\`](https://github.com/kh4f/relion/commit/4343e6f)

##### &emsp;&ensp;&nbsp;&nbsp; [_Release Changelog_](https://github.com/kh4f/relion/blob/main/CHANGELOG.md) &ensp;‚Ä¢&ensp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0) &ensp;‚Ä¢&ensp; _Sep 12, 2025_


"
`;

exports[`partials customization > should generate changelog with transformed partials 1`] = `
"# Changelog


CUSTOM HEADER---## &ensp; [\` üì¶ v0.8.0  \`](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0)

---### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ö†Ô∏è BREAKING CHANGES
- The \`defaultChangelogSections\` export has been removed from the public API. Use \`changelogSectionsSelector\` instead, which provides the same sections plus \`pick\` and \`omit\` methods. <sup>[1]</sup>
- The \`sections\` option in changelog configuration is now an object with section IDs as keys instead of an array. <sup>[2]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;‚ú® Features
- **utils**: add \`changelogSectionsSelector\` with \`pick\` and \`omit\` methods [\`3febcbc\`](https://github.com/kh4f/relion/commit/3febcbc) ‚ö†Ô∏è<sup>[1]</sup>

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üöú Refactoring
- **changelog**: 
  - change \`sections\` from array to object format for better usability [\`349fed9\`](https://github.com/kh4f/relion/commit/349fed9) ‚ö†Ô∏è<sup>[2]</sup>
  - rename \`commitGroups\` property to \`commitTypeGroups\` [\`6ec89a8\`](https://github.com/kh4f/relion/commit/6ec89a8)

### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;üè∑Ô∏è Types
- **defaults**: use \`satisfies ChangelogSectionsMap\` for \`defaultChangelogSections\` [\`4343e6f\`](https://github.com/kh4f/relion/commit/4343e6f)

CUSTOM FOOTER---##### &emsp;&ensp;&nbsp;&nbsp; [_All Release Commits_](https://github.com/kh4f/relion/compare/v0.7.0...v0.8.0) &ensp;‚Ä¢&ensp; _Sep 12, 2025_


---"
`;
